### PR TITLE
SQL: Return Intervals in SQL format for CLI

### LINKED ISSUE #29970: SQL: ensure proper conversion of values when we format output in CLI
*Original comment by @imotov:*

Currently we convert values using Objects.toString() that doesn't work for dates that are represented as longs and might not work for other value types as well. Converted NOCOMMITs LINK REDACTED and LINK REDACTED.

### LINKED ISSUE #36432: SQL: CLI returns Intervals in ISO instead of SQL format
The CLI doesn't parse the given strings, rather returns them as they are which, for intervals, means the internal format (ISO) is used instead of the desired one (SQL).
The formatting code gets called however since the strings are not parsed, they don't match either as date or interval.

### PR DESCRIPTION
**Pull request by astefan**
This PR adds a new CLI mode and removes the CLI `client_id`. Uses the new Mode to decide what format to use for Intervals in the responses to CLI requests. Fixes https://github.com/elastic/elasticsearch/issues/36432.

The PR also fixes https://github.com/elastic/elasticsearch/issues/29970 by not making any conversions, since the received data is already formatted.

Another small improvement in this PR is the use of the right Mode for cursor close requests.

Also, a nice consequence of these changes is that the CLI will not greatly depend on the future SQL protocol changes, but it will only display what receives from the server, paving the way for safe changes like https://github.com/elastic/elasticsearch/issues/36186.

### CODE REVIEW

File: x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/SqlQueryResponse.java [hunk 1]
```diff
@@ -173,7 +175,12 @@ public static XContentBuilder value(XContentBuilder builder, Mode mode, Object v
             ZonedDateTime zdt = (ZonedDateTime) value;
             // use the ISO format
             builder.value(StringUtils.toString(zdt));
-        } else {
+        } else if (mode == CLI && value != null && value.getClass().getSuperclass().getSimpleName().equals(INTERVAL_CLASS_NAME)) {
```
Comment (matriv) [inline x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/SqlQueryResponse.java]: I think it's better to use the `isAssignableFrom()` and avoid the string check.

**Review by matriv — APPROVED** [review]
LGTM. Left a comment

**Comment by astefan** [comment]
run the gradle build tests 1

File: x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/SqlQueryResponse.java (hunk 1 above)
Comment (astefan) [reply x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/SqlQueryResponse.java]: Agree, but `Interval` is not visible in the `action` project.

**Comment by astefan** [comment]
run the gradle build tests 2

**Comment by astefan** [comment]
:-(( run the gradle build tests 1

**Comment by astefan** [comment]
run the gradle build tests 1

File: x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/SqlQueryResponse.java (hunk 1 above)
Comment (matriv) [reply x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/SqlQueryResponse.java]: ah ok, thx!

**Comment by astefan** [comment]
run the gradle build tests 1

**Comment by astefan** [comment]
run the gradle build tests 1

**Comment by astefan** [comment]
run the gradle build tests 1

**Comment by costin** [comment]
run the gradle build tests 1

**Review by costin — APPROVED** [review]
LGTM. The mode has changed but I didn't see any update on the metrics side - is it counted for REST?

**Review by costin — APPROVED** [review]
LGTM2

**Review by matriv — APPROVED** [review]
LGTM!

**Pull request merged**